services/galley/src/Galley/API/Util.hs

```diff
@@ -57,29 +57,30 @@ ensureConnectedOrSameTeam u uids = do
   sameTeamUids <- forM uTeams $ \team ->
     fmap (view userId) <$> Data.teamMembersLimited team uids
   -- Do not check connections for users that are on the same team
-  ensureConnected u (makeIdOpaque <$> uids \\ join sameTeamUids)
+  ensureConnectedToLocals u (uids \\ join sameTeamUids)
```

Comment:
with 'locals' we here mean the "local version of the remote id" right?

Reply:
No, `ensureConnectedToLocals` only accepts local user IDs. It checks the connections of users on the same backend ("locals" or maybe "local users"?), just by looking at the database.

For now it is sufficient to call this function here, since `ensureConnectedOrSameTeam` is only called with local IDs, but maybe it's cleare to use `ensureConnected u (Local <$> _)` to show that in the future we will have to consider remote users, too? I could also bubble up the type by already making `ensureConnectedOrSameTeam` accept MappedOrLocalId and converting at the call sites. What do you think?

Reply:
Add a note to future us is probably sufficient for now